base/broadcast.jl

```diff
@@ -234,7 +234,10 @@ end
 end
 
 # broadcast methods that dispatch on the type found by inference
-function broadcast_t(f, ::Type{Any}, shape, iter, As...)
+function broadcast_t(f, ::Type{Any}, T::Type, shape, iter, As...)
+    if isempty(iter)
+        return similar(Array{T}, shape)
+    end
```

Comment:
Ups. I was playing around reorganizing the code and left this, but shouldn't be necessary. I'll put it back as it was.